clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -3981,23 +4348,13 @@ private void prepareJoinAndSyncResponse(String consumerId, int generation, List<
         }, syncGroupResponse(assignment, Errors.NONE));
     }
 
-    private Map<TopicPartition, Errors> partitionErrors(Collection<TopicPartition> partitions, Errors error) {
-        final Map<TopicPartition, Errors> errors = new HashMap<>();
-        for (TopicPartition partition : partitions) {
-            errors.put(partition, error);
-        }
-        return errors;
-    }
-
-    private void respondToOffsetCommitRequest(final Map<TopicPartition, Long> expectedOffsets, Errors error) {
-        Map<TopicPartition, Errors> errors = partitionErrors(expectedOffsets.keySet(), error);
-        client.respond(offsetCommitRequestMatcher(expectedOffsets), offsetCommitResponse(errors));
-    }
-
-    private MockClient.RequestMatcher offsetCommitRequestMatcher(final Map<TopicPartition, Long> expectedOffsets) {
+    private MockClient.RequestMatcher offsetCommitRequestMatcher(
+        Map<TopicPartition, Long> expectedOffsets,
+        Consumer<OffsetCommitRequest> requestConsumer
```

Comment:
From my reading of the code this consumer is a captor. We validate some of the things in this method and we validate the overall captured value elsewhere in individual tests. I am not too certain how this can be simplified to just a matcher to be honest.

Reply:
@dajac Sure, that makes sense to get rid of the consumer, since it is mixing test design pattern and overlap the responsibilities of the matcher as you pointed out.